docs/manual/introduction/Optimizations.html

```diff
@@ -0,0 +1,179 @@
+            [page:Object3D.matrixAutoUpdate] is set true, which is the default case this could have a very bad impact.
+        </p>
+
+        <p>
+            If we have a look at the current implementation of the updateMatrixWorld. When [page:Object3D.matrixAutoUpdate]
+            is true then all children matrices will be force updated.
+        </p>
+
+        <code>
+updateMatrix: function () {
+
+    this.matrix.compose( this.position, this.quaternion, this.scale );
+
+    this.matrixWorldNeedsUpdate = true;
+
+},
+
+updateMatrixWorld: function ( force ) {
+
+    if ( this.matrixAutoUpdate ) this.updateMatrix();
+
+    if ( this.matrixWorldNeedsUpdate || force ) {
+
+        if ( this.parent === null ) {
+
+            this.matrixWorld.copy( this.matrix );
+
+        } else {
+
+            this.matrixWorld.multiplyMatrices( this.parent.matrixWorld, this.matrix );
+
+        }
+
+        this.matrixWorldNeedsUpdate = false;
+
+        force = true;
+
+    }
+
+    // update children
+
+    var children = this.children;
+
+    for ( var i = 0, l = children.length; i < l; i ++ ) {
+
+        children[ i ].updateMatrixWorld( force );
+
+    }
+
+},
+        </code>
+
+        <p>
+            In order to deal with this you can set [page:Scene.autoUpdate] false and manage your world matrices by calling
+            [page:Object3D.updateMatrixWorld] when you update a property.
+            You can also set [page:Object3D.matrixAutoUpdate] and call [page:Object3D.updateMatrix] before calling any
+            [page:Object3D.updateMatrixWorld].
+        </p>
+
+		<h2>Textures</h2>
+
+        <p>
+            Textures can have a very big CPU and memory overhead. Three.js requires textures to be power of two. When
+            three.js encounter a not power of two image, it will resize it which consume quite a lot CPU.
+        </p>
+
+        <p>
+            You also need to take care of the device capabilities and especially the max texture size given by
+            [page:WebGLRenderer.capabilities.maxTextureSize]. If your texture is higher than that then three.js will
+            resize it.
+        </p>
+
+		<h2>Shadows</h2>
+
+        <p>
+            By default [page:WebGLRenderer.shadowMap.autoUpdate] is set true which will cause the shadow to update at
+            each frame. If you know your scene is not intended to change at each frame, you can disable this and use
+            [page:WebGLRenderer.shadowMap.needsUpdate] in order to trigger a recompute.
+        </p>
+
+        <h2>Power Preference</h2>
+
+        <p>
+            On devices with 2 GPU, generally integrated and dedicated the browser will by default use the integrated one
+            in order to boost battery performance.
+
+            If your scene requires a lot of performances, you can choose to use the dedicated GPU by using the
+            **powerPreference** parameter of [page:WebGLRenderer].
```

Comment:
This is not quite accurate, the spec says that it's a "hint to the useragent" and up to browsers to decide whether to implement this. As far as I know, at least on Windows none of them do.

Reply:
Ok, I wasn't aware of this, I will learn a lot with that PR :smile: